packages/expo-av/ios/EXAV/Video/EXVideoView.m

```diff
@@ -129,15 +139,19 @@ - (void)_removePlayer
     _requestedFullscreenChange = NO;
   }
 
-  if (_data) {
-    [_data pauseImmediately];
-    [_data setStatusUpdateCallback:nil];
-    [_exAV demoteAudioSessionIfPossible];
-    [self _removeFullscreenPlayerViewController];
-    [self _removePlayerLayer];
-    [self _removePlayerViewController];
-    _data = nil;
-  }
+  // Any ViewController/layer updates need to be
+  // executed on the main UI thread.
+  __weak EXVideoView *weakSelf = self;
+  void (^block)(void) = ^ {
+    __strong EXVideoView *strongSelf = weakSelf;
+    if (strongSelf) {
+      [strongSelf _removeFullscreenPlayerViewController];
+      [strongSelf _removePlayerLayer];
+      [strongSelf _removePlayerViewController];
+    }
+  };
```

Comment:
```suggestion
  UM_WEAKIFY(self);
  void (^block)(void) = ^ {
    UM_ENSURE_STRONGIFY(self);
    [self _removeFullscreenPlayerViewController];
    [self _removePlayerLayer];
    [self _removePlayerViewController];
  };
```

Reply:
Good one, thanks for the reminder! I'll create a PR to update this globally for expo-av 👍

Reply:
Created a PR for this 👍 
https://github.com/expo/expo/pull/9437